src/homelogstore/log_store.cpp

```diff
@@ -352,8 +356,10 @@ nlohmann::json HomeLogStore::dump_log_store(const log_dump_req& dump_req) {
             }
             json_records.emplace_back(std::move(json_val));
             decltype(idx) end_idx{std::min(max_idx, dump_req.end_seq_num)};
-            end_iterate = (cur_idx < end_idx) ? true : false;
-            return end_iterate;
+            proceed = (cur_idx < end_idx && --batch_size > 0) ? true : false;
+            // User can provide either the end_seq_num or batch_size in the request.
+            if (cur_idx < end_idx && batch_size == 0) { json_dump["next_cursor"] = std::to_string(cur_idx + 1); }
```

Comment:
if batch_size hasn't reached to zero yet, but cur_idx has advanced to be as equal to end_idx, we should also set "next_cursor", right?

Reply:
I think what we need is just:

if (!proceed) {...}